examples/string-templates/string_templates.bal

```diff
@@ -0,0 +1,26 @@
+import ballerina/io;
+
+public function main() {
+    // Note how the first `\n` is treated as is.
```

Comment:
Can we start with a simple string template first? And then have examples of special cases.